src/CommonRuntime/IO.fs

```diff
@@ -203,7 +213,11 @@ let private watchForChanges (uri:Uri) (((tp:IDisposableTypeProvider), typeName)
 
         if (match typeNameBeingDisposedOpt with None -> true | Some typeNameBeingDisposed -> typeName = typeNameBeingDisposed) then 
             if watcher.Remove tp typeName then
+#if FX_NO_CONCURRENT
                 watchers.Remove uri.OriginalString |> ignore
+#else
+                watchers.TryRemove(uri.OriginalString) |> ignore
+#endif
             
```

Comment:
Can you please pu the watchers.add and watchers.Remove inside a lock on watchers in the FX_NO_CONCURRENT branch?